src/modules/avformat/producer_avformat.c

```diff
@@ -1332,20 +1429,48 @@ static int convert_image( producer_avformat self, AVFrame *frame, uint8_t *buffe
 	}
 	else
 	{
+		int i, c;
+		AVPicture output;
+		struct sliced_pix_fmt_conv_t ctx =
+		{
+			.flags = flags,
+			.width = width,
+			.height = height,
+			.frame = frame,
+			.output = &output,
+			.dst_format = AV_PIX_FMT_YUYV422,
+			.src_colorspace = self->yuv_colorspace,
+			.dst_colorspace = profile->colorspace,
+			.src_full_range = self->full_luma,
+			.dst_full_range = 0,
+		};
 #if defined(FFUDIV) && (LIBAVFORMAT_VERSION_INT >= ((55<<16)+(48<<8)+100))
-		struct SwsContext *context = sws_getContext( width, height, src_pix_fmt,
-			width, height, AV_PIX_FMT_YUYV422, flags | SWS_FULL_CHR_H_INP, NULL, NULL, NULL);
+		ctx.src_format = src_pix_fmt;
 #else
-		struct SwsContext *context = sws_getContext( width, height, pix_fmt,
-			width, height, AV_PIX_FMT_YUYV422, flags | SWS_FULL_CHR_H_INP, NULL, NULL, NULL);
+		ctx.src_format = pix_fmt;
 #endif
-		AVPicture output;
-		avpicture_fill( &output, buffer, AV_PIX_FMT_YUYV422, width, height );
-		if ( !set_luma_transfer( context, self->yuv_colorspace, profile->colorspace, self->full_luma, 0 ) )
-			result = profile->colorspace;
-		sws_scale( context, (const uint8_t* const*) frame->data, frame->linesize, 0, height,
-			output.data, output.linesize);
-		sws_freeContext( context );
+		ctx.src_desc = av_pix_fmt_desc_get( ctx.src_format );
+		ctx.dst_desc = av_pix_fmt_desc_get( ctx.dst_format );
+
+		avpicture_fill( ctx.output, buffer, ctx.dst_format, width, height );
+
+		if ( getenv("MLT_AVFORMAT_SLICED_PIXFMT") )
```

Comment:
> I don't have a lot of experience with the SWS API. What is the cost of allocating a new context for each slice? Is it expensive?

i think it expensive, but it get a performance gain: https://github.com/mltframework/mlt/issues/172#issuecomment-278329812

> Would it be possible to create a single context for each frame and then have each slice use the same context?

i though about it, but it will require some code to check if sws init condition changes

> Would it be possible to create a single context on initialization and use the same context for all slices and all frames?

i tried that - it is not working.

> Do we want this to be an option? If it works, why not always uses it?

for first time if some artifact appear on a picture we could swap processing method to check if it caused by sliced processing...